ocaml/libs/xapi-inventory/lib/inventory.ml

```diff
@@ -52,10 +52,12 @@ let inventory = Hashtbl.create 10
 
 let inventory_m = Mutex.create ()
 
+let new_uuid = Uuidm.v4_gen (Random.get_state ())
```

Comment:
We can only ever do this once in the entire program`: `Random.get_state` returns a *copy* of the current state.
This means that `new_uuid` from this file, and the one from Vhd_format will generate the same UUIDs, because they started out with the same random state.

It would be better to do what Uuidm did previously, which is to call `Random.State.make_self_init ()`, which returns a state that is not shared with anyone.

(or since 5.0 we could also use Random.split, but not in 4.x)